relay-server/src/actors/spooler/mod.rs

```diff
@@ -977,6 +1000,28 @@ mod tests {
         ManagedEnvelope::untracked(envelope, outcome_aggregator, test_store)
     }
 
+    #[tokio::test]
+    async fn create_spool_directory_path() {
+        let parent_dir = std::env::temp_dir().join(Uuid::new_v4().to_string());
+        let target_dir = parent_dir
+            .join("subdir1")
+            .join("subdir2")
+            .join("spool-file");
+        let buffer_guard: Arc<_> = BufferGuard::new(1).into();
+        let config: Arc<_> = Config::from_json_value(serde_json::json!({
+            "spool": {
+                "envelopes": {
+                    "path": target_dir,
+                }
+            }
+        }))
+        .unwrap()
+        .into();
+        BufferService::create(buffer_guard, services(), config)
+            .await
+            .unwrap();
```

Comment:
Could we actually also add assert there to check that the folders and the path to the file exist?